### PR TITLE
Functional Graphs and IO

### PR DESCRIPTION
**Pull request by mayankumar04**
Just added some java support to this section

### CODE REVIEW

**Comment by mayankumar04** [comment]
the bot didn't recognize the imports from the java library, ill remove those

**Comment by dongliu0426** [comment]
Please use the live editor (http://usaco.guide/liveupdate/).

**Comment by frodakcin** [comment]
Will review when deployment is successful. Like dongliu0426 said, try using the live editor to debug.

**Comment by mayankumar04** [comment]
alr i finally debugged it and the tests were green

**Comment by frodakcin** [comment]
nice!

File: content/1_General/IO.mdx [hunk 1]
```diff
@@ -69,6 +69,8 @@ It doesn't matter. Whichever you're most comfortable with!
 
 ### Method 1: [<iostream\>](http://www.cplusplus.com/reference/iostream/)
 
+This library is quite straight forward to use and the `cin` function simply assigns values sepearated by spaces or lines to different variables. Similarly, the `cout` function simply writes on the standard console and either an escape sequence (that's what the `\n` does below) or the `endl` command can enter a new lijne at the end.
```
Comment (frodakcin) [inline content/1_General/IO.mdx:72]: "lijne" typo

File: content/1_General/IO.mdx [hunk 2]
```diff
@@ -487,6 +489,29 @@ int main() {
 }
 ```
 
+### Method 3: Re-defining IO
+
+The standard IO functions can also be redefined to work with file IO. The following template (which is commonly used on USACO solution pages) uses this redefinition to solve this example. After re-defining the IO functions, then you can simply use `cin` and `cout` to read and write data.
```
Comment (frodakcin) [inline content/1_General/IO.mdx:494]: State that you are redefining using https://en.cppreference.com/w/cpp/io/c/freopen

File: content/1_General/IO.mdx [hunk 3]
```diff
@@ -495,6 +520,24 @@ int main() {
 
 Again, `BufferedReader` and `PrintWriter` should be used. The provided constructor `public Kattio(String problemName)` handles file input / output for you.
 
+```java
```
Comment (frodakcin) [inline content/1_General/IO.mdx:523]: Mention Kattio further down? Doesn't really make sense that it says "provided constructor `public Kattio(String problemName)`" here when it's provided much later.

**Comment by HarryWangATX** [comment]
i'll review this soon thx for fixing!

**Review by frodakcin — CHANGES_REQUESTED** [review]
Minor edits in IO, Func graph checks out for me.

File: content/1_General/IO.mdx (hunk 3 above)
Comment (mayankumar04) [reply content/1_General/IO.mdx:523]: Ok i don't understand why include kattio at all, literally no one will use it in cp. they'll just use bufferedreader or scanner

**Review by HarryWangATX — COMMENTED** [review]
Looks good! Just follow revise with @frodakcin 's suggestions

**Comment by mayankumar04** [comment]
alr i made the edits that @frodakcin recommended

**Comment by dongliu0426** [comment]
@mayankumar04 you can re-request their review

File: content/1_General/IO.mdx (hunk 3 above)
Comment (frodakcin) [reply content/1_General/IO.mdx:523]: it's template that's faster to copy-paste than to write out bufferedreader/printwriter

File: content/1_General/IO.mdx [hunk 4]
```diff
@@ -495,6 +520,24 @@ int main() {
 
 Again, `BufferedReader` and `PrintWriter` should be used. The provided constructor `public Kattio(String problemName)` handles file input / output for you.
```
Comment (frodakcin) [inline content/1_General/IO.mdx:521]: I think the sentence "The provided constructor `public Kattio(String problemName)` handles file input / output for you." can just be deleted. Kattio is explained below already.

**Review by frodakcin — CHANGES_REQUESTED** [review]
One final thing. Should be good to merge after

**Comment by mayankumar04** [comment]
The newly edited file removes a dead link

**Review by frodakcin — APPROVED** [review]

**Pull request merged**